Refresh installing and calling contracts

ipopescu opened the pull request:
### Related links

- #349
- #363 
- https://github.com/casper-ecosystem/counter/pull/19

### Changes

- This is a major refresh of the "calling contracts" page, focusing on using the Casper command-line client. 
- I split the "installation" and the "calling contracts" sections into 2 pages because the page became too long.

### Notes

- I tested all the examples with the [Counter Contract](https://github.com/casper-ecosystem/counter) and submitted a PR to refresh that contract as well: https://github.com/casper-ecosystem/counter/pull/19.
- I also added examples related to the erc20 contract.
- The pages ran locally without issues.
- Another card (#393) is open for documenting the same steps via Rust code.
- This note was moved to card #393 - "Note: This result is returned as bytes. These bytes need to be deserialized into the correct CLType in a smart contract or a dApp. Refer to [casper-types](https://docs.rs/casper-types/latest/casper_types/bytesrepr/index.html) to find the latest APIs for deserialization."

ACStoneCL commented on source/docs/casper/dapp-dev-guide/installing-contracts.md:
```diff
@@ -0,0 +1,321 @@
+   - Your environment meets these [prerequisites](/workflow/setup/) and you have a client to interact with the network, such as the [default Casper client](/workflow/setup#the-casper-command-line-client)
+   - You have a [Casper account](/workflow/setup/#setting-up-an-account) with a public and secret key pair to initiate the deploy
+   - You have enough CSPR tokens in your account to pay for deploys. If you plan to use the Casper Testnet, learn about the [faucet](/workflow/token-transfer#2-the-faucet) to fund your testing account
+- You understand how to write basic contract code and session code
```
```suggestion
- You understand how to [write basic contract code](/dapp-dev-guide/writing-contracts/introduction.md) and session code
```

Adding a link to the Writing Smart Contracts intro might not be necessary, but it could help. I think once the session code page is live, that could also be linked to the 'session code' text here.

ipopescu replied:
Yes, absolutely. Thank you! I'll add a reminder to add in the session code link later.

ACStoneCL reviewed (approved):
Just one added link suggestion that is definitely optional. Content looks great!

ramujin commented on source/docs/casper/dapp-dev-guide/calling-contracts.md:
```diff
@@ -1,132 +1,310 @@
+
+## Calling Versioned Contracts by Name {#calling-versioned-contracts-by-name}
+
+You can access an entry point in a specific contract version by referencing the contract name, entry point, and version number. When creating the contract using [new_contract](https://docs.rs/casper-contract/latest/casper_contract/contract_api/storage/fn.new_contract.html), provide the `hash_name`, which puts the [ContractHash](https://docs.rs/casper-types/latest/casper_types/contracts/struct.Con ...
```
To what is the "hash_name" referring?

ipopescu replied:
It is the contract package name stored under the account's named keys. I have reworked this section to make this clearer. I am still referring to hash_name as it is defined in the new_contract API. Commit a6270cb4b71e68409f70d7b2cd89829b3ff7d469.

ramujin commented on source/docs/casper/dapp-dev-guide/calling-contracts.md:
````diff
@@ -1,132 +1,310 @@
 ```
 
-## Calling a Contract by Hash and Entry Point {#calling-a-contract-by-hash-and-entry-point}
+This example code stores the "contract_name" into a NamedKey, which you can reference once you install the contract in global state. In this case, the ContractHash will be stored under the "contract_name" NamedKey.
````
Where is "contract_name"?

ipopescu replied:
This was a mistake, I was supposed to say counter_package_name. I have corrected it with commit a6270cb4b71e68409f70d7b2cd89829b3ff7d469.

ramujin reviewed (commented):
Looks good overall! I just had a few minor comments based on confusion that I had while reading through it.

ramujin commented on source/docs/casper/dapp-dev-guide/installing-contracts.md:
```diff
@@ -0,0 +1,321 @@
+
+The arguments used above are:
+-   `node-address` - An IP address of a peer on the network. The default port for JSON-RPC servers on Mainnet and Testnet is 7777
+-   `state-root-hash` - 
```
This is missing a description for state-root-hash

ramujin reviewed (commented):
I had forgotten one comment from the installing-contracts file.

ipopescu commented on source/docs/casper/dapp-dev-guide/calling-contracts.md:
```diff
@@ -1,132 +1,310 @@
+Smart contracts exist as stored on-chain logic, allowing disparate users to call the included entry points. This tutorial covers different ways to call Casper contracts with the [Casper command-line client](/workflow/setup/#the-casper-command-line-client) and the `put-deploy` command.
 
-First, set up the contract name so you can call it using the name in subsequent deploys. The following code sample uses `sample_contract` as the contract name.
+**IMPORTANT**: You will receive a deploy hash when calling the `put-deploy` command. You need this hash to verify that the deploy executed successfully.
```
Yes, knowing how to send/verify deploys is a prerequisite, so I've placed it afterward with commit 24a678226da1e9d4c503147e6da59f9672f9126e.

ipopescu commented:
Remaining items in this PR:
- Can I refer to the counter and erc20 contracts? Or should I remove the links and leave only the sample code?
- Finally, I need an example for this section "Calling Contracts that Return a Value" using the Rust client

ipopescu commented:
@darthsiroftardis I've addressed your feedback with commits 61de534d39ec3a947ad27d2264149e5a590a90ae, 43cb0acc29a6dfda807214488634f1b19b6cd5b4 and 6d5b8500141d97e70be308d5d6587999c1347c8e.
